src/core/tests/Test_CoreMain.py

```diff
@@ -1134,6 +1159,624 @@ def __check_telemetry_events(self, runtime):
+            substatus_file_data = json.load(file_handle)
+
+        self.assertTrue(len(json.dumps(substatus_file_data)) < Constants.StatusTruncationConfig.AGENT_FACING_STATUS_FILE_SIZE_LIMIT_IN_BYTES)
+        self.__assert_complete_status_output(substatus_file_data, Constants.ASSESSMENT, Constants.PATCH_ASSESSMENT_SUMMARY, Constants.STATUS_SUCCESS, patch_counts + 3)
+
+        # Assert truncated status file
+        with runtime.env_layer.file_system.open(runtime.execution_config.status_file_path, 'r') as file_handle:
+            substatus_file_data = json.load(file_handle)
+
+        self.assertTrue(len(json.dumps(substatus_file_data)) < Constants.StatusTruncationConfig.AGENT_FACING_STATUS_FILE_SIZE_LIMIT_IN_BYTES)
+        substatus_file_data = substatus_file_data[0]["status"]["substatus"][0]
+        self.assertEqual(substatus_file_data["name"], Constants.PATCH_ASSESSMENT_SUMMARY)
+        self.assertNotEqual(substatus_file_data["status"], Constants.STATUS_WARNING.lower())
+        self.assertEqual(len(json.loads(substatus_file_data["formattedMessage"]["message"])["patches"]), patch_counts + 3)
+        # status_file_patches = json.loads(substatus_file_data["formattedMessage"]["message"])["patches"]
+        # self.assertNotEqual(status_file_patches[1]['patchId'], "Truncated_patch_list_id")
+        # self.assertTrue('additional updates of classification' not in status_file_patches[1]['name'])
+        self.assertEqual(json.loads(substatus_file_data["formattedMessage"]["message"])["errors"]["code"], 0)
+        self.assertEqual(len(json.loads(substatus_file_data["formattedMessage"]["message"])["errors"]["details"]), 0)
+        self.assertFalse("review this log file on the machine" in json.loads(substatus_file_data["formattedMessage"]["message"])["errors"]["message"])
+        # self.assertEqual(len(runtime.status_handler._StatusHandler__assessment_packages_removed), 0)
+
+        runtime.stop()
+
+    def test_assessment_operation_truncation_over_size_limit(self):
+        argument_composer = ArgumentComposer()
+        argument_composer.operation = Constants.ASSESSMENT
+        runtime = RuntimeCompositor(argument_composer.get_composed_arguments(), True, Constants.ZYPPER)
+        runtime.set_legacy_test_type('HappyPath')
+        CoreMain(argument_composer.get_composed_arguments())
+
+        # Check telemetry events
+        self.__check_telemetry_events(runtime)
+
+        # HappyPath add 3 additional packages
+        # {\"patchId\": \"kernel-default_4.4.49-92.11.1_Ubuntu_16.04\", \"name\": \"kernel-default\", \"version\": \"4.4.49-92.11.1\", \"classifications\": [\"Security\"]},
+        # {\"patchId\": \"libgcc_5.60.7-8.1_Ubuntu_16.04\", \"name\": \"libgcc\", \"version\": \"5.60.7-8.1\", \"classifications\": [\"Other\"]},
+        # {\"patchId\": \"libgoa-1_0-0_3.20.5-9.6_Ubuntu_16.04\", \"name\": \"libgoa-1_0-0\", \"version\": \"3.20.5-9.6\", \"classifications\": [\"Other\"]}
+
+        patch_counts = random.randint(780, 1000)
```

Comment:
patch_count - singular  (applies anywhere such a variable name is used)

Reply:
addressed

Reply:
Checked and resolving